miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebChromeClient.kt

```diff
@@ -21,8 +31,41 @@ internal class MiniAppWebChromeClient(context: Context) : WebChromeClient() {
         super.onReceivedTitle(webView, title)
     }
 
+    // region geolocation
+    private var geoLocationRequestOrigin: String? = null
+    @VisibleForTesting
+    internal var geoLocationCallback: GeolocationPermissions.Callback? = null
+
+    @Suppress("FunctionMaxLength")
+    override fun onGeolocationPermissionsShowPrompt(
+        origin: String?,
+        callback: GeolocationPermissions.Callback?
+    ) {
+        geoLocationRequestOrigin = origin
+        geoLocationCallback = callback
+        if (ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION)
+            == PackageManager.PERMISSION_GRANTED)
+            onGeolocationPermissionResult(true)
+        else
+            miniAppMessageBridge.requestPermission(
+                MiniAppPermission.getPermissionRequest(MiniAppPermission.PermissionType.GEOLOCATION),
+                MiniAppPermission.Code.GEOLOCATION
+            )
+    }
+
+    override fun onGeolocationPermissionResult(isGranted: Boolean) {
+        geoLocationCallback?.invoke(geoLocationRequestOrigin, isGranted, isGranted)
+        geoLocationRequestOrigin = null
+        geoLocationCallback = null
+    }
+    // end region
+
     @VisibleForTesting
     internal fun doInjection(webView: WebView) {
         webView.evaluateJavascript(bridgeJs) {}
     }
 }
+
+internal interface WebChromeListener {
```

Comment:
Would it sound reasonable to update `WebChromeListener` -> `WebChromePermissionListener` ?